gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/TimeBasedSubDirDatasetsFinder.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.compaction.mapreduce;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import com.google.common.collect.Sets;
+
+import gobblin.compaction.Dataset;
+import gobblin.configuration.State;
+import gobblin.util.DatasetFilterUtils;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * A {@link DatasetsFinder} based on subdir of the inputDir.
+ * Each subfolder in {@link #inputDir} is considered a dataset, if it satisfies blacklist and whitelist.
+ *
+ * For example, suppose {@link #inputDir} is '/inputdata', and a topic is
+ * '/inputdata/mytopic'. If this topic is considered a {@link Dataset}, all data under this folder will be compacted.
+ * If a particular day's data of this topic is considered a {@link Dataset}, e.g.,
+ * '/inputdata/mytopic/daily/2015/08/09', then data under this folder will be compacted.
+ */
+@Slf4j
+public class TimeBasedSubDirDatasetsFinder extends DatasetsFinder {
```

Comment:
Since this class is called `TimeBasedSubDirDatasetsFinder`, Most of the logic in `MRCompactorTimeBasedJobPropCreator` should be moved here, such as parsing the folder path to get folder time, check whether folder time is within allowed period, etc.